apiserver/facades/client/machinemanager/machinemanager.go

```diff
@@ -372,7 +373,29 @@ func (mm *MachineManagerAPI) destroyMachine(args params.Entities, force, keep bo
 				logger.Warningf("could not keep instance for machine %v: %v", machineTag.Id(), err)
 			}
 		}
-		var info params.DestroyMachineInfo
+		info := params.DestroyMachineInfo{
```

Comment:
destroyMachine is already over 150 lines long. Since you're here, please split out the embedded function and even it's pieces into more manageable bits.

Reply:
Just realised I still need to do this

Reply:
I've split out some of the more awkward bits of this function. it's still about 100 lines long, but that seems to be the smallest I can reasonably break it up